### PR TITLE
Backend implementation of filtering environments by cluster ARN

### PR DESCRIPTION
**Pull request by poojamaiya**
#### Summary
Implements the backend piece for filtering environments by cluster ARN

#### Implementation details
Lists all environments and then filters by looking at the cluster field.

#### Testing

- [X] Binary built locally and unit-tests pass (`make`)  
- [X] Build in docker succeeds (`make release`)

New tests cover the changes: Yes

#### Description for the changelog
Backend implementation of filtering environments by cluster ARN

#### Licensing

This contribution is under the terms of the Apache 2.0 License: Yes

### CODE REVIEW

File: daemon-scheduler/pkg/deployment/environment.go [hunk 1]
```diff
@@ -135,6 +141,36 @@ func (e environment) ListEnvironments(ctx context.Context) ([]types.Environment,
...
+		return nil, errors.New("Filter key is missing")
+	}
+	if filterVal == "" {
+		return nil, errors.New("Filter value is missing")
+	}
+	switch filterKey {
+	case clusterFilter:
+		return e.filterEnvironmentsByCluster(ctx, filterVal)
+	default:
+		return nil, errors.New("Unsupported filter key")
```
Comment (kiranmeduri) [inline daemon-scheduler/pkg/deployment/environment.go]: Also better to echo back the key provided

File: daemon-scheduler/pkg/deployment/environment.go [hunk 2]
```diff
@@ -22,6 +22,10 @@ import (
 	"github.com/pkg/errors"
 )
 
+const (
+	clusterFilter = "cluster"
```
Comment (kiranmeduri) [inline daemon-scheduler/pkg/deployment/environment.go]: can this be a enum?

File: daemon-scheduler/pkg/deployment/test_constants.go [hunk 3]
```diff
@@ -20,7 +20,8 @@ const (
 	taskDefinition   = "arn:aws:ecs:us-east-1:12345678912:task-definition/test"
 	taskARN1         = "arn:aws:ecs:us-east-1:12345678912:task/c024d145-093b-499a-9b14-5baf273f5835"
 	taskARN2         = "arn:aws:ecs:us-east-1:12345678912:task/a1d71628-01e3-4013-b18c-6e14032a9522"
-	cluster          = "arn:aws:ecs:us-east-1:123456789123:cluster/test"
```
Comment (kiranmeduri) [inline daemon-scheduler/pkg/deployment/test_constants.go]: why not just leave this in here and add cluster1 and cluster2? no big deal though

File: daemon-scheduler/pkg/deployment/environment_test.go [hunk 4]
```diff
@@ -203,6 +203,38 @@ func (suite *EnvironmentTestSuite) TestListEnvironments() {
...
+}
+
+func (suite *EnvironmentTestSuite) TestFilterEnvironmentsUnsupportedFilterKey() {
+	suite.environmentStore.EXPECT().ListEnvironments(gomock.Any()).Times(0)
+
+	_, err := suite.environment.FilterEnvironments(suite.ctx, "unsupportedFilter", "filterVal")
+	assert.Error(suite.T(), err, "Expected an error when filter key is unsupported")
+}
+
+func (suite *EnvironmentTestSuite) TestFilterEnvironmentsByClusterARN() {
```
Comment (kiranmeduri) [inline daemon-scheduler/pkg/deployment/environment_test.go]: can you also add a test for the case where ListEnvironments returning error.

File: daemon-scheduler/pkg/deployment/environment_test.go [hunk 5]
```diff
@@ -203,6 +203,38 @@ func (suite *EnvironmentTestSuite) TestListEnvironments() {
...
+	allEnvs := []types.Environment{*suite.environment1, *suite.environment2}
```
Comment (kiranmeduri) [inline daemon-scheduler/pkg/deployment/environment_test.go]: can we have multiple envs of cluster1 to check that all are returned.

File: daemon-scheduler/pkg/deployment/environment.go (hunk 1 above)
Comment (poojamaiya) [inline daemon-scheduler/pkg/deployment/environment.go]: Will do

File: daemon-scheduler/pkg/deployment/environment.go (hunk 2 above)
Comment (poojamaiya) [inline daemon-scheduler/pkg/deployment/environment.go]: Whats the advantage?

File: daemon-scheduler/pkg/deployment/test_constants.go (hunk 3 above)
Comment (poojamaiya) [inline daemon-scheduler/pkg/deployment/test_constants.go]: Felt weird.. I'm going to leave in the changes and not undo them unless you feel strongly.

File: daemon-scheduler/pkg/deployment/environment_test.go (hunk 4 above)
Comment (poojamaiya) [inline daemon-scheduler/pkg/deployment/environment_test.go]: Will do

File: daemon-scheduler/pkg/deployment/environment_test.go (hunk 5 above)
Comment (poojamaiya) [inline daemon-scheduler/pkg/deployment/environment_test.go]: Will do!

**Pull request merged**